src/Troopers/BehatContexts/Context/ExtendedEntityContext.php

```diff
@@ -39,4 +40,88 @@ public function beforeScenario($event)
             }
         }
     }
+
+    /**
+     * @Then /^I should find (\d+) (.*) like:$/
+     *
+     * @param $nbr
+     * @param $name
+     * @param TableNode $table
+     *
+     * @throws \Exception
+     */
+    public function existObjectLikeFollowing($nbr, $name, TableNode $table)
+    {
+        $entityName = $this->resolveEntity($name)->getName();
+
+        $rows = $table->getRows();
+        $headers = array_shift($rows);
+
+        $row = array_shift($rows);
+        $objects = $this->getEntityManager()
+            ->getRepository($entityName)
+            ->findBy(
+                $this->getQueryParams($entityName, $headers, $row)
+            );
+
+        if (count($objects) === 0) {
+            throw new \Exception(sprintf('There is not any %s for the following params: %s', $name, json_encode($this->getQueryParams($entityName, $headers, $row))));
```

Comment:
`$this->getQueryParams($entityName, $headers, $row)` is called 3 times in this method, a variable may be used to store this data.